operators/src/pro/cache/cache_chunks.rs

```diff
@@ -391,31 +320,337 @@ impl_cache_result_check!(MultiPolygon);
+pub type CompressedMultiPointCollection = CompressedFeatureCollection<MultiPoint>;
+pub type CompressedMultiLineStringCollection = CompressedFeatureCollection<MultiLineString>;
+pub type CompressedMultiPolygonCollection = CompressedFeatureCollection<MultiPolygon>;
+
+#[derive(Debug)]
+pub struct CompressedFeatureCollectionImpl<G, C> {
+    data: Vec<u8>,
+    spatial_bounds: Option<geoengine_datatypes::primitives::BoundingBox2D>,
+    time_interval: Option<geoengine_datatypes::primitives::TimeInterval>,
+    types: HashMap<String, geoengine_datatypes::primitives::FeatureDataType>,
+    cache_hint: geoengine_datatypes::primitives::CacheHint,
+    collection_type: std::marker::PhantomData<G>,
+    compression_type: std::marker::PhantomData<C>,
+}
+
+impl<G, C> ByteSize for CompressedFeatureCollectionImpl<G, C> {
+    fn heap_byte_size(&self) -> usize {
+        self.data.heap_byte_size()
+    }
+}
+
+impl<G, C> CompressedFeatureCollectionImpl<G, C>
+where
+    FeatureCollection<G>: FeatureCollectionInfos + CacheElementSpatialBounds,
+    C: ChunkCompression,
+{
+    pub fn from_collection(collection: FeatureCollection<G>) -> Result<Self, CacheError> {
+        let spatial_bounds = collection.cache_element_spatial_bounds();
+        let time_interval = collection.time_bounds();
+
+        let FeatureCollectionInternals {
+            table,
+            types,
+            collection_type,
+            cache_hint,
+        } = collection.into();
+
+        let data = C::compress_array(table)?;
+
+        let compressed: CompressedFeatureCollectionImpl<G, C> = CompressedFeatureCollectionImpl {
+            data,
+            spatial_bounds,
+            time_interval,
+            types,
+            cache_hint,
+            collection_type,
+            compression_type: Default::default(),
+        };
+
+        Ok(compressed)
+    }
+
+    pub fn to_collection(&self) -> Result<FeatureCollection<G>, CacheError> {
+        let table = C::decompress_array(&self.data)?;
+
+        let collection_internals = FeatureCollectionInternals {
+            table,
+            types: self.types.clone(),
+            collection_type: self.collection_type,
+            cache_hint: self.cache_hint,
+        };
+
+        let collection = FeatureCollection::from(collection_internals);
+
+        Ok(collection)
+    }
+}
+
+pub trait ChunkCompression {
+    fn compress_array(collection: StructArray) -> Result<Vec<u8>, CacheError>
+    where
+        Self: Sized;
+
+    fn decompress_array(compressed: &[u8]) -> Result<StructArray, CacheError>
+    where
+        Self: Sized;
+
+    fn array_to_bytes(table: StructArray) -> Result<Vec<u8>, CacheError> {
+        let record_batch = RecordBatch::from(&table);
+
+        let mut file_writer = FileWriter::try_new_with_options(
+            Vec::new(),
+            record_batch.schema().as_ref(),
+            IpcWriteOptions::default(), // TODO: Add ".try_with_compression(COMPRESSION)?," once issues are resolved
```

Comment:
maybe specify the "issues" for future generations

Reply:
[21fe3b6](https://github.com/geo-engine/geoengine/pull/864/commits/21fe3b6b20248c32d049f0099b08426b18ff1767)